packages/liferay-npm-scripts/README.md

```diff
@@ -79,19 +78,42 @@ liferay-npm-scripts theme build
 
 Runs the "build" task, providing it with the configuration it needs to find core dependencies such as the [`liferay-frontend-theme-styled` base theme files](https://github.com/liferay/liferay-portal/tree/master/modules/apps/frontend-theme/frontend-theme-styled/src/main/resources/META-INF/resources/_styled).
 
-### eject
+## Config
 
-**Note: this is a one-way operation. Once you eject, you can't go back.**
+> Note: as of v2.x the config file was renamed from `.liferaynpmscriptsrc` to `liferaynpmscripts.js`
```

Comment:
I would be concerned with that being too generic and might cause people to think that is a file geared specifically to liferay's build process, when really its only specific to the npm tools.

What about one of these?
`npmscripts`
`npm-scripts`
`liferayscripts`
`liferay-scripts`

and then a file ending of either `.config.js` or `.conf.js`

I think `npm-scripts.config.js` is pretty clean

Reply:
The hyphens definitely help readability (`.liferaynpmscriptsrc` was terrible in that respect). To be honest, any of the suggested names are an improvement. I'd add one more to the list (`liferay-npm-scripts.config.js`) for consideration too.

Reply:
I like the new options better, but I think I still want to try to win you over to `build.js`.

This is an aggregation of different options found in a `build.gradle` file:

```groovy
buildService {
	apiDir = "../adaptive-media-image-api/src/main/java"
	testDir = "../adaptive-media-image-test/src/testIntegration/java"
}

dependencies {
	compileOnly group: "com.liferay.portal", name: "com.liferay.portal.kernel", version: "default"
	compileOnly project(":apps:static:portal-configuration:portal-configuration-metatype-api")

	testCompile group: "com.liferay.portal", name: "com.liferay.portal.impl", version: "default"
	testCompile project(":core:petra:petra-lang")

	testIntegrationCompile group: "com.liferay.portal", name: "com.liferay.portal.kernel", version: "default"
	testIntegrationCompile project(":apps:adaptive-media:adaptive-media-api")
}

deploy {
	enabled = false
}

liferayOSGi {
	expandCompileInclude = true
}
```

If you look into it, it basically is declaring dependencies and overriding gradle task defaults. It seems awfully close to what we're getting to now, only that we will be doing this for "npm scripts (tasks)".

This is indeed tied to liferay build process and that's the initial philosophy. We're actually already forcing everyone to use `liferay-npm-scripts build` for their frontend build mechanisms (exclusions aside). We also currently only have 4-5 places where we need to override defaults (and that number might shrink), so it shouldn't be that bad.

What do you think?

Reply:
Personally I just think `build.js` is too generic and doesn't give any idea that there is a dependency required. To push back on the build.gradle, `gradle` is fairly intrinsic to Liferay's build system and require to do *any*(as far as I know...) deployment. Whereas, `liferay-npm-scripts` is an additional tool we can add to our module. If `liferay-npm-scripts` worked without having to declare it as a dependency in our package.json, then I could see the argument for `build.js` but since `liferay-npm-scripts` is an *extra*(required to be listed as dependency) then I think its necessary to make the config explicit in its naming.

Additionally, the extension `.gradle` really helps with the naming of the file. If it was `build.java` I think it would be much more confusing. Since `.js` is so generic and used for a variety of things, I really think it helps keeping the naming explicit.

In the end though, you're the boss, so I am willing to do whatever naming convention you want. 🤷‍♂

Reply:
> If liferay-npm-scripts worked without having to declare it as a dependency in our package.json, then I could see the argument for build.js but since liferay-npm-scripts is an extra(required to be listed as dependency) then I think its necessary to make the config explicit in its naming.

Technically, it is only listed as a dependency on the root package (which is not really a package), but in the places where it is used it's not, it is a given of the system, just like Gradle

> Additionally, the extension .gradle really helps with the naming of the file. If it was build.java I think it would be much more confusing. Since .js is so generic and used for a variety of things, I really think it helps keeping the naming explicit.

This is my biggest concern too...

> In the end though, you're the boss, so I am willing to do whatever naming convention you want. 🤷‍♂

I really really hope we can do better than that :)

What would you think about `build.npmscripts` or `build.lfrnpmscripts` or something along those lines?

Reply:
I'd be down for `build.npmscripts`, although it can cause issue for linters and language identification in editors since it doesnt have the .js extension. So I think that would mean these files would sneak past the SF.

> I really really hope we can do better than that :)

hey, bikeshedding needs a kill switch at some point in the end 🚴

Reply:
> I'd be down for build.npmscripts, although it can cause issue for linters and language identification in editors since it doesnt have the .js extension. So I think that would mean these files would sneak past the SF.

So... `build.npmscripts.js`?

> hey, bikeshedding needs a kill switch at some point in the end 🚴

That's true too, but then we should at least put it to a vote 😂

Reply:
`build.npmscripts.js` sounds good to me, although I did just remember that inside the config we have keys of `lint` `format` and `build`. Do you think that'll be confusing at all to put lint and format configs inside of a file named build?

Reply:
Maybe a little, but shouldn't be too much... if you draw the connection, you're just providing  configuration to different tasks.

We currently only have... 4 or 5 instances of this and all of them use it to configure build dependencies. We might not even allow getting out of linting or formatting much, so that might never be a real thing.

I'd be fine with this, but we can wait for @wincent's opinion on this for a final decision. 

If `build.npmscripts.js` is confusing, then the other only options would be the ones you proposed at the beginning. I like this one better for the simmetry it brings, which is always appreciated imho.

Reply:
I don't feel super strongly about this so I wasn't going to add any more ⛽️ to the 🔥, but seeing as you asked...

You both make good points. I agree about needing the ".js" extension, and I agree that "build" is a bit misleading for a tool that does things beyond just building. That's why I liked the explicitness of "liferay-npm-scripts.config.js" because it very obviously shows that it is "configuration for liferay-npm-scripts". I don't think that "build.npmscripts.js" is quite as clear, but it is definitely an improvement over "build.js". Another concern with `build.js` is it sounds like a script that I should be able to run (eg. `node build.js`) whereas the variants which end with ".config.js" are much more clearly not intended to be run directly.

So my order of preference would be:

1. liferay-npm-scripts.config.js
2. npm-scripts.config.js
3. npmscripts.config.js
4. build.npmscripts.js
5. build.js
6. .liferaynpmscriptsrc

Reply:
Ok, taking from @wincent's list, I would remove the ones with `-` because it's a complicated topic in the portal repo, so, let's go with `npmscripts.config.js` and be done with this :)